sql/connect/client/jdbc/src/main/scala/org/apache/spark/sql/connect/client/jdbc/util/JdbcTypeUtils.scala

```diff
@@ -37,6 +37,7 @@ private[jdbc] object JdbcTypeUtils {
     case StringType => Types.VARCHAR
     case _: DecimalType => Types.DECIMAL
     case DateType => Types.DATE
+    case BinaryType => Types.BINARY
```

Comment:
`VARBINARY` might be more suitable here.

some references:

- Databricks JDBC driver uses BINARY for BinaryType
- PostgreSQL JDBC driver uses BINARY for bytea
- Trino JDBC driver uses 
- MariaDB JDBC driver uses VARBINARY/LONGVARBINARY for TINYBLOB/MEDIUMBLOB/LONGBLOB/BLOB

Reply:
thanks for the comment, I'll raise a followup PR